WordPress/src/main/java/org/wordpress/android/ui/stories/prefs/StoriesPrefs.kt

```diff
@@ -0,0 +1,129 @@
+package org.wordpress.android.ui.stories.prefs
+
+import android.content.Context
+import android.net.Uri
+import androidx.preference.PreferenceManager
+import com.wordpress.stories.compose.story.StoryFrameItem
+import com.wordpress.stories.compose.story.StoryFrameItem.BackgroundSource.FileBackgroundSource
+import com.wordpress.stories.compose.story.StoryFrameItem.BackgroundSource.UriBackgroundSource
+import com.wordpress.stories.compose.story.StorySerializerUtils
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.LocalId
+import org.wordpress.android.fluxc.model.LocalOrRemoteId.RemoteId
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class StoriesPrefs @Inject constructor(
+    private val context: Context
+) {
+    companion object {
+        private val KEY_PREFIX_STORIES_SLIDE_ID = "story_slide_id-"
+        private val KEY_PREFIX_LOCAL_MEDIA_ID = "l-"
+        private val KEY_PREFIX_REMOTE_MEDIA_ID = "r-"
+    }
+
+    private fun buildSlideKey(siteId: Long, mediaId: RemoteId): String {
+        return KEY_PREFIX_STORIES_SLIDE_ID + siteId.toString() + "-" +
+                KEY_PREFIX_REMOTE_MEDIA_ID + mediaId.value.toString()
+    }
+
+    private fun buildSlideKey(siteId: Long, mediaId: LocalId): String {
+        return KEY_PREFIX_STORIES_SLIDE_ID + siteId.toString() + "-" +
+                KEY_PREFIX_LOCAL_MEDIA_ID + mediaId.value.toString()
+    }
+
+    fun checkSlideIdExists(siteId: Long, mediaId: RemoteId): Boolean {
+        val slideIdKey = buildSlideKey(siteId, mediaId)
+        return PreferenceManager.getDefaultSharedPreferences(context).contains(slideIdKey)
+    }
+
+    fun checkSlideOriginalBackgroundMediaExists(siteId: Long, mediaId: RemoteId): Boolean {
+        val storyFrameItem: StoryFrameItem? = getSlideWithRemoteId(siteId, mediaId)
+        storyFrameItem?.let { frame ->
+            // now check the background media exists or is accessible on this device
+            frame.source.let { source ->
+                if (source is FileBackgroundSource) {
+                    source.file?.let {
+                        return it.exists()
+                    } ?: return false
+                } else if (source is UriBackgroundSource) {
+                    source.contentUri?.let {
+                        return isUriAccessible(it)
+                    } ?: return false
+                }
+            }
+        }
+        return false
+    }
+
+    private fun isUriAccessible(uri: Uri): Boolean {
```

Comment:
I can't help but think that this probably doesn't belong here 😅 I know we have `UriUtils`  in `WordPressUtils` so I am wondering if it's best to put it there? You would lose the kotlin goodness 😉 but it would be in a place where it's reusable. WDYT?

Reply:
That's certainly something that crossed my mind as well, but given it also involves some administrativia with other repos and is not particularly needed anywhere else (can you think of a real use case where this would be needed elsewhere in WP?), I think it's fine to keep it here where there's a meaning to checking for whether the background source is accessible or not (it really helps the decision making about allowing editing or not in our case). If you'd like to, we can open an issue in the meantime instead so to focus on the rest of PRs, does that make sense? 🙏